src/redux/slices/dashboard.ts

```diff
@@ -1,22 +1,29 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
-import { VisualisationMappingFunctionPath } from '../../utils/visualisationMapping';
 import { DashboardVisualisation } from '../../types/DashboardVisualisation';
+import { v4 as uuidv4 } from 'uuid';
 
 export default createSlice({
     name: 'Dashboard Items',
-    initialState: {} as Record<VisualisationMappingFunctionPath, DashboardVisualisation>,
+    initialState: {} as Record<string, DashboardVisualisation>,
     reducers: {
         add: (state, action: PayloadAction<DashboardVisualisation>) => {
-            const key = `${action.payload.dataSourceId}-${action.payload.visualisationType}` as VisualisationMappingFunctionPath;
-            state[key] = action.payload;
+            const key = `${uuidv4()}`;
```

Comment:
If it's a string, use `uuidv4()`, and if it can be casted to it, use `uuidv4() as string`, and if should be converted, use `String(uuidv4)`. These are all more explicit (or in the former case, implicit) and should be preferred to templates.